File: Python/ProgramacaoOrientadaAObjetos-Python/Introducao_POO/Aluno.py
#Classe:
class Alunoo:
    
    notaA1 = 0
    notaA2 = 0
    media = 0
    frequencia = 0
        
#Métodos:
def set_nome(self, nome):
    
    #Condicionando o NOME à ser uma String
    if(type(nome)==type(str())):
        self._nome = nome
    else:
        print('Nome deve ser uma String!')
    
def set_ra(self, ra):
    #Condicionando o R.A à ser um INT
    if(type(ra)==type(int())):
        self._ra =ra
    else:
        print('R.A deve ser um Número Inteiro!')
        
def set_notaA1(self, notaA1):
    self.notaA1 = notaA1
    
def set_notaA2(self, notaA2):
    self.notaA2 = notaA2
    
def calcular_Media(self):
    self.media = (self.notaA1 + (2*self.notaA2)) / 3

def set_frequencia(self, freq):
    self.frequencia = freq
    
def get_media(self):
    print('A média do aluno ', self._aluno, 'é: ', self.media)
    
def situacao_Aluno(self):
    if(self.frequencia < 75):
        print('Aluno está reprovado por faltas!')
        
    elif(self.media >= 5):
        print('Aluno está reprovado!')
        
    elif(self.media >=3):
        print('Aluno está de RE!')
        
    else:
        print('Aluno está reprovado por nota!')
        
def getDadosAluno(self):
    print('\nAluno: ', self._nome)
    print('R.A: ', self._ra)
    print('Frequência: ', self.frequencia)
    print('Nota A1: ', self.notaA2)
    print('Nota A2: ', self.notaA2)
    print('Média: ', self.media)
    
    
    #Atributo PRIVADO:
    _nome = property(set_nome)
    _ra = property(set_ra)

File: Java/Estudando/Aula02_ex02/src/Aluno.java

public class Aluno {

	private String nome;
	private int ra;
	private float notaA1;
	private float notaA2;
	private float media;
	private float frequencia;
	
	//Métodos:
	public void setNome(String n) {
		this.nome = n;
	} 
	
	public void setRa(int ra) {
		this.ra = ra;
	} 
	
	public void setNotaA1(float n1) {
		this.notaA1 = n1;
	} 
	
	public void setNotaA2(float n2) {
		this.notaA2 = n2;
	} 
	
	public void CalcularMedia() {
		this.media = (this.notaA1 + (2*this.notaA2)) / 3;
	}
	
	public void setFrequencia(float freq) {
		this.frequencia = freq;
	}
	
	public float getFrequencia() {
		return frequencia;
	}
		
	public void getMedia() {
		System.out.println("A média do aluno " + this.nome + " é: " + String.format("%.2f", this.media));
	}
	
	public void situacaoAluno() {
		if(this.frequencia < 75) {
			System.out.println("Aluno está reprovado por faltas!");
		}
		else if(this.media >= 5) {
			System.out.println("Aluno está aprovado!");
		}
		else if(this.media >= 3) {
			System.out.println("Aluno está de RE!");
		}
		else {
			System.out.println("Aluno está reprovado por nota!");
		}
	}
	
	public void getDadosAluno() {
		System.out.println("\n");
		System.out.println("Aluno: " + this.nome);
		System.out.println("R.A: " + this.ra);
		System.out.println("Frequencia: " + this.frequencia);
		System.out.println("Nota A1: " + this.notaA1);
		System.out.println("Nota A2: " + this.notaA2);
		System.out.println("Média: " + String.format("%.2f", this.media));
	}
	
	
}
